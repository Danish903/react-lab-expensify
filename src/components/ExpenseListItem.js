import React from 'react';

const Item = ({ description, amount, createdAt }) => (
  <div>
    <h4>{description}</h4>
    <p>
      ${amount} - {createdAt}
    </p>
  </div>
);

export default Item;
