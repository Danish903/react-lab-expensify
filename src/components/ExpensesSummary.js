import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({
  expensesCount,
  hiddenExpensesCount,
  expensesTotal
}) => (
  <div className="page-header">
    <div className="content-container">
      <h1 className="page-header__title">
        Viewing <span>{expensesCount}</span>{' '}
        {expensesCount > 1 ? 'expenses' : 'expense'} totalling{' '}
        <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span>
      </h1>
      {hiddenExpensesCount === 1 && <h3>1 expense filtered...</h3>}
      {hiddenExpensesCount > 1 && (
        <h3>{hiddenExpensesCount} expenses filtered...</h3>
      )}
      <div className="page-header__actions">
        <Link className="button" to="/create">
          Add Expenses
        </Link>
      </div>
    </div>
  </div>
);
const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    hiddenExpensesCount: state.expenses.length - visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
