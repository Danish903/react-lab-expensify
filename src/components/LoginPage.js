import React from 'react';
import { connect } from 'react-redux';
import { startGoogleLogin, startGithubLogin } from '../actions/auth';

export const LoginPage = props => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>Get your expenses under control!</p>
      <div>
        <button
          className="button button--login"
          onClick={props.startGoogleLogin}
        >
          Login with Google
        </button>
        <button
          className="button button--login"
          onClick={props.startGithubLogin}
        >
          Login with Github
        </button>
      </div>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startGithubLogin: () => dispatch(startGithubLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
