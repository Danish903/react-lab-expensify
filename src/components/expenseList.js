import React from 'react';

const ExpenseList = () => (
  <div>
    <h3> Expense List</h3>
  </div>
);

export default ExpenseList;
