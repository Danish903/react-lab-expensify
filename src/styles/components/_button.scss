.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}

.button--login {
  margin:0 0 $s-size 0; 
  flex:1;
}
