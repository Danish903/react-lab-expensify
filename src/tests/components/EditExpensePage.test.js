import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixures/expenses';

let editExpense;
let removeExpense;
let history;
let wrapper;

beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <EditExpensePage
      expense={expenses[2]}
      editExpense={editExpense}
      removeExpense={removeExpense}
      history={history}
    />
  );
});

test('should render EditExpensePage', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
  expect(editExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[1]);
  expect(history.push).toHaveBeenLastCalledWith('/');
});

test('should handle removeExpense', () => {
  wrapper.find('button').simulate('click');
  expect(removeExpense).toHaveBeenLastCalledWith(expenses[2].id);
  expect(history.push).toHaveBeenLastCalledWith('/');
});
