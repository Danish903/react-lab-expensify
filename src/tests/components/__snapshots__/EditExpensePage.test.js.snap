// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 3000,
        "createdAt": 432000000,
        "description": "Gas Bill",
        "id": "3",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
